manifests/config/server/service.pp

```diff
@@ -37,19 +41,22 @@
   }
 
   if $service_ensure =~ /^(absent|false)$/ {
-    $changes = "rm Server/Service[#attribute/name='${name}']"
+    $changes = "rm Server/Service[#attribute/name='${service_name}']"
   } else {
-    if $class_name_ensure =~ /^(absent|false)$/ {
-      $_class_name = "rm Server/Service[#attribute/name='${name}']/#attribute/className"
-    } elsif $class_name {
-      $_class_name = "set Server/Service[#attribute/name='${name}']/#attribute/className ${class_name}"
+    if $class_name_ensure =~ /^(present|true)$/ {
+      if empty($class_name) {
+        fail('$class_name must be specified when $class_name_ensure is set to true or present')
+      }
+      $_class_name = "set Server/Service[#attribute/name='${service_name}']/#attribute/className ${class_name}"
+    } else {
+      $_class_name = "rm Server/Service[#attribute/name='${service_name}']/#attribute/className"
```

Comment:
The rest of the PR looks good to me, but why this backwards-incompatible change? As far as I can tell, anyone passing `class_name => 'whatever.class.name',` would get the class name added, but after your change they would also have to add `class_name_ensure => 'present'` as well, else it will be removed. Could you undo that part of this PR? Thanks!